gomock/matchers.go

```diff
@@ -168,6 +169,33 @@ func (n notMatcher) String() string {
 	return "not(" + n.m.String() + ")"
 }
 
+type regexMatcher struct {
+	regex      *regexp.Regexp
+	compileErr error
+}
+
+func (m regexMatcher) Matches(x any) bool {
+	if m.regex == nil {
+		return false
+	}
+
+	switch t := x.(type) {
+	case string:
+		return m.regex.MatchString(t)
+	case []byte:
+		return m.regex.Match(t)
+	default:
+		return false
+	}
+}
+
+func (m regexMatcher) String() string {
+	if m.compileErr != nil {
+		return m.compileErr.Error()
+	}
```

Comment:
```suggestion
type regexMatcher struct {
	regex *regexp.Regexp
}

func (m regexMatcher) Matches(x any) bool {
	switch t := x.(type) {
	case string:
		return m.regex.MatchString(t)
	case []byte:
		return m.regex.Match(t)
	default:
		return false
	}
}

func (m regexMatcher) String() string {
```
Rationale: `mock` is for testing and it should be okay to panic if the regular expression is invalid. The current code silently ignores invalid regular expressions, which in my opinion is not the best for testing.

Note: this suggestion must be accepted with the suggested changes to `Regex` below.

Reply:
Good point, a panic is probably preferable here